spec-0013/index.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "SPEC 13 — Recommended targets and naming conventions"
+number: 13
+date: 2024-06-05
+author:
+  - "Pamphile Roy <roy.pamphile@gmail.com>"
+  - "Matthias Bussonnier <>"
+  - "Jarrod Millman <millman@berkeley.edu>"
+discussion: https://discuss.scientific-python.org/t/spec-13-recommended-targets-and-naming-conventions
+endorsed-by:
+---
+
+## Description
+
+For consistency and decreased cognitive load across the ecosystem, this SPEC recommends naming conventions around various project aspects--such as project structure, repository layout, folder names, task runner and `pyproject.toml` targets name.
```

Comment:
I would also add some defaults for declaring optional dependencies: `all`, `test`, and `docs`.

Reply:
`test` -> `tests`? 🤔

Reply:
never mind, should have read all the comments below before commenting, as this was already mentioned.

Reply:
`dev` would also be nice.

Reply:
test --> tests, bike shedding, but I'm 👎 on this. As it's test-dependencies in my view while the other recommendation is about a directory that contains the tests

Reply:
Just hard to remember that "s" is for testing dir (`pytest tests/`) but no "s" for install (`pip install .[test]`).

Reply:
astropy does it this way since forever 😄

Reply:
Not too fond of `[dev]`, this makes people think it's okay to install all your development dependencies into a single environment, which is not true. Also not fond of `[all]`, same sort of reason. I've seen `[all]` used well for projects that have several optional dependencies and then provide an `[all]`, but it's not useful to add things like `[test]` dependencies.

Reply:
We also limited the discussion to `test[s]` and `doc[s]` and did not discuss other target to now grow the spec too much. 

We can extend to other naming conventions later.